### PR TITLE
UCL-436 rename Snackbar to Alert

### PR DESCRIPTION
**Pull request by akosszeredai**
This is a member of:
https://onewelcome.atlassian.net/browse/UCL-356

### CODE REVIEW

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx [hunk 1]
```diff
@@ -123,21 +121,24 @@ export const SnackbarItem = ({
...
+      onMouseLeave={onItemBlur}
+      onFocus={onItemFocus}
+      onBlur={onItemBlur}
+      role="alert"
     >
       <div className={classes["outer-content-wrapper"]}>
         <Icon icon={getVariantIcon()} className={classes["icon"]} />
-        <div className={classes["content-wrapper"]}>
-          {renderTitle && (
+        <div className={classes["content-wrapper"]} role="log">
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertItem/AlertItem.tsx:133]: no need for `role='log'` https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/log_role

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx [hunk 2]
```diff
@@ -123,21 +121,24 @@ export const SnackbarItem = ({
...
       aria-live="polite"
-      className={snackbarClasses}
-      onMouseEnter={onMouseEnter}
-      onMouseLeave={onMouseLeave}
+      className={alertClasses}
+      onMouseEnter={onItemFocus}
+      onMouseLeave={onItemBlur}
+      onFocus={onItemFocus}
+      onBlur={onItemBlur}
+      role="alert"
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertItem/AlertItem.tsx:129]: I'd remove `role='alert'`.

Setting role="alert" is equivalent to setting [aria-live="assertive"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) and [aria-atomic="true"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-atomic)

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx [hunk 3]
```diff
@@ -123,21 +121,24 @@ export const SnackbarItem = ({
...
     >
       <div className={classes["outer-content-wrapper"]}>
         <Icon icon={getVariantIcon()} className={classes["icon"]} />
-        <div className={classes["content-wrapper"]}>
-          {renderTitle && (
+        <div className={classes["content-wrapper"]} role="log">
+          {!!title && !!content && (
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertItem/AlertItem.tsx:134]: please bring back renderTitle

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx [hunk 4]
```diff
@@ -123,21 +121,24 @@ export const SnackbarItem = ({
...
             <Typography className={classes["title"]} variant="body-bold" tag="span">
               {title}
             </Typography>
           )}
-          {renderContentOrTitleOnly && (
+          {(!!content || !!title) && (
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertItem/AlertItem.tsx:139]: please bring back `renderContentOrTitleOnly`

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx [hunk 5]
```diff
@@ -123,21 +121,24 @@ export const SnackbarItem = ({
...
             <Typography className={classes["content"]} variant="body">
-              {hasOnlyTitle ? title : content}
+              {content || title}
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertItem/AlertItem.tsx:141]: I'd bring back `hasOnlyTitle`.

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx [hunk 6]
```diff
@@ -123,21 +121,24 @@ export const SnackbarItem = ({
...
       ref={ref}
       aria-live="polite"
-      className={snackbarClasses}
-      onMouseEnter={onMouseEnter}
-      onMouseLeave={onMouseLeave}
+      className={alertClasses}
+      onMouseEnter={onItemFocus}
+      onMouseLeave={onItemBlur}
+      onFocus={onItemFocus}
+      onBlur={onItemBlur}
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertItem/AlertItem.tsx:128]: good catch!

File: src/components/Notifications/Alert/AlertProvider/AlertProvider.tsx [hunk 7]
```diff
@@ -57,7 +57,9 @@ export interface Item {
   onClose?: () => void;
 }
 
-export const SnackbarProvider = (
+export type PropsOrTitle = EnqueueAlertProps | string | undefined;
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/AlertProvider/AlertProvider.tsx:60]: Move it to interfaces file.

File: src/components/Notifications/NotificationProvider/NotificationContext.test.tsx [hunk 8]
```diff
@@ -71,29 +71,29 @@ const renderSnackbarProvider = () => {
   };
 
   const queries = render(
-    <SnackbarProvider closeButtonTitle="close">
+    <AlertProvider closeButtonTitle="close" key={5}>
```
Comment (pawel-napieracz) [inline src/components/Notifications/NotificationProvider/NotificationContext.test.tsx:74]: key is useless

**Review by pawel-napieracz — CHANGES_REQUESTED** [review]

File: src/components/Notifications/Alert/useAlert.ts [hunk 9]
```diff
@@ -15,15 +15,15 @@
...
-  const ctx = useContext(SnackbarContext);
+export const useAlert = () => {
+  const ctx = useContext(AlertContext);
 
   return {
-    enqueueWarningSnackbar: ctx.enqueueWarningSnackbar,
-    enqueueErrorSnackbar: ctx.enqueueErrorSnackbar,
-    enqueueSuccessSnackbar: ctx.enqueueSuccessSnackbar,
-    enqueueSnackbar: ctx.enqueueSnackbar
+    enqueueWarningAlert: ctx.enqueueWarningAlert,
```
Comment (pawel-napieracz) [inline src/components/Notifications/Alert/useAlert.ts:24]: please keep backward compatibility

File: src/index.ts [hunk 10]
```diff
@@ -98,8 +98,8 @@ export type {
 } from "./components/Notifications/NotificationProvider/notification.interfaces";
 export { SlideInModal } from "./components/Notifications/SlideInModal/SlideInModal";
 export type { Props as SlideInModalProps } from "./components/Notifications/SlideInModal/SlideInModal";
-export { SnackbarProvider } from "./components/Notifications/Snackbar/SnackbarProvider/SnackbarProvider";
-export type { Props as SnackbarProviderProps } from "./components/Notifications/Snackbar/SnackbarProvider/SnackbarProvider";
+export { AlertProvider } from "./components/Notifications/Alert/AlertProvider/AlertProvider";
+export type { Props as AlertProviderProps } from "./components/Notifications/Alert/AlertProvider/AlertProvider";
```
Comment (pawel-napieracz) [inline src/index.ts:102]: please keep backward compatibility

File: src/index.ts [hunk 11]
```diff
@@ -72,7 +72,7 @@ export { generateID, debounce, throttle } from "./util/helper";
 /* Notifications */
 export { Modal } from "./components/Notifications/Modal/Modal";
 export type { Props as ModalProps } from "./components/Notifications/Modal/Modal";
-export { useSnackbar } from "./components/Notifications/Snackbar/useSnackbar";
+export { useAlert } from "./components/Notifications/Alert/useAlert";
```
Comment (pawel-napieracz) [inline src/index.ts:75]: please keep backward compatibility

**Review by pawel-napieracz — CHANGES_REQUESTED** [review]

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx (hunk 1 above)
Comment (akosszeredai) [reply src/components/Notifications/Alert/AlertItem/AlertItem.tsx:133]: Removed in the visuals branch.

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx (hunk 2 above)
Comment (akosszeredai) [reply src/components/Notifications/Alert/AlertItem/AlertItem.tsx:129]: Default is polite. Error medium and high emphasis gets the assertive.

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx (hunk 3 above)
Comment (akosszeredai) [reply src/components/Notifications/Alert/AlertItem/AlertItem.tsx:134]: Reason?

File: src/components/Notifications/Alert/AlertItem/AlertItem.tsx (hunk 6 above)
Comment (akosszeredai) [reply src/components/Notifications/Alert/AlertItem/AlertItem.tsx:128]: Ty

**Pull request closed without merging**